instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/aspects/WithSpanAspectTest.java

```diff
@@ -187,4 +202,132 @@ void suppressServerSpan() throws Throwable {
                 trace.hasSpansSatisfyingExactly(
                     parentSpan -> parentSpan.hasName("parent").hasKind(SERVER)));
   }
+
+  @Nested
+  @DisplayName("with a method annotated with @WithSpan returns CompletionStage")
+  class WithCompletionStage {
+
+    @Test
+    @DisplayName("should end Span on complete")
+    void onComplete() throws Throwable {
+      CompletableFuture<String> future = new CompletableFuture<>();
+
+      // when
+      withSpan("parent", () -> withSpanTester.testAsyncCompletionStage(future));
+
+      try {
+        instrumentation.waitForTraces(1, 20, TimeUnit.SECONDS);
```

Comment:
So this test will always take at least 20 seconds? Thus this test class will take more than a minute to complete. We have to come up with some other way :(

Reply:
Agreed, I need to find a better way to assert the negative here and would love some pointers.  Some of the Groovy tests seem to sleep for 500ms and I'll be happy to follow the pattern.

Actually this test is currently "failing" as the parent span is recorded, which I wasn't expecting, so I'll either need to stop running it under the parent span or find a better way to assert that only the parent span has completed.